### PR TITLE
Update for DnD5E v2.1.2

### PR DESCRIPTION
**Pull request by aquavitae**
This pulls together fixes for at least #624, #625 and #628. I'm not completely sure that deleting the blocks of code as described in #625 is the right way of doing it, but it seems to work so unless anyone knows better I'll leave it as is for now.

### CODE REVIEW

File: src/scripts/tidy5e-npc.js [hunk 1]
```diff
@@ -30,10 +30,10 @@ export default class Tidy5eNPC extends dnd5e.applications.actor
    * @return {Object}
    */
   static get defaultOptions() {
-    let defaultTab = game.settings.get("tidy5e-sheet", "defaultActionsTab") != 'default' 
+    let defaultTab = game.settings.get("tidy5e-sheet", "defaultActionsTab") != 'default'
```
Comment (szefo09) [inline src/scripts/tidy5e-npc.js:33]: Don't add different end of line sequences

File: src/scripts/tidy5e-npc.js [hunk 2]
```diff
@@ -30,10 +30,10 @@ export default class Tidy5eNPC extends dnd5e.applications.actor
...
       ? game.settings.get("tidy5e-sheet", "defaultActionsTab")
       : 'attributes' ;
-		if (!game.modules.get('character-actions-list-5e')?.active && 
+		if (!game.modules.get('character-actions-list-5e')?.active &&
```
Comment (szefo09) [inline src/scripts/tidy5e-npc.js:36]: Don't add different end of line sequences

File: src/scripts/tidy5e-sheet.js [hunk 3]
```diff
@@ -915,7 +915,7 @@ Hooks.on('renderAbilityUseDialog', (application, html, context) => {
                 availableTextSlotsFounded = lastMatch;
               }
             }
-            
+
```
Comment (szefo09) [inline src/scripts/tidy5e-sheet.js:918]: Fix this tabulation

File: src/templates/actors/tidy5e-sheet.html [hunk 4]
```diff
@@ -15,12 +15,12 @@
 				</article>
 			</div>
 		</section>
-
```
Comment (szefo09) [inline src/templates/actors/tidy5e-sheet.html:18]: Fix this tabulation

File: src/templates/actors/tidy5e-sheet.html [hunk 5]
```diff
@@ -15,12 +15,12 @@
...
 		</section>
-    
+
 		{{> "dnd5e.actor-warnings"}}
 		{{!-- Sheet Header --}}
 		<header class="tidy5e-header">
-			
-			
+
+
```
Comment (szefo09) [inline src/templates/actors/tidy5e-sheet.html:23]: Fix this tabulation

File: src/templates/actors/tidy5e-sheet.html [hunk 6]
```diff
@@ -259,7 +259,7 @@ <h4 class="ability-name box-title rollable" title="{{ability.label}}">{{ability.
 			</div>
 
 		</header>
-		
+
```
Comment (szefo09) [inline src/templates/actors/tidy5e-sheet.html:262]: Fix this tabulation

File: src/templates/actors/tidy5e-sheet.html [hunk 7]
```diff
@@ -279,16 +279,16 @@ <h4 class="ability-name box-title rollable" title="{{ability.label}}">{{ability.
 		<section class="sheet-body">
 
 			<div class="tab attributes" data-group="primary" data-tab="attributes">
-				
+
```
Comment (szefo09) [inline src/templates/actors/tidy5e-sheet.html:282]: Fix this tabulation

File: src/templates/actors/tidy5e-sheet.html [hunk 8]
```diff
@@ -315,11 +315,11 @@ <h4 class="skill-name rollable">{{skill.label}}</h4>
 									<h4>{{ localize "TIDY5E.RestoreOnRest" }}</h4>
 									<input id="{{@root/appId}}-{{res.name}}.sr" name="system.resources.{{res.name}}.sr" type="checkbox" {{checked res.sr}}>
 									<label for="{{@root/appId}}-{{res.name}}.sr" class="checkbox" title="{{ localize 'TIDY5E.RestS' }}">
-										{{ localize "DND5E.RestS" }} 
+										{{ localize "DND5E.RestS" }}
```
Comment (szefo09) [inline src/templates/actors/tidy5e-sheet.html:318]: Don't change End of Line Sequence here.

File: src/templates/actors/tidy5e-sheet.html [hunk 9]
```diff
@@ -315,11 +315,11 @@ <h4 class="skill-name rollable">{{skill.label}}</h4>
...
 									</label>
 									<input id="{{@root/appId}}-{{res.name}}.lr" name="system.resources.{{res.name}}.lr" type="checkbox" {{checked res.lr}}>
 									<label for="{{@root/appId}}-{{res.name}}.lr" class="checkbox" title="{{ localize 'TIDY5E.RestL' }}">
-										{{ localize "DND5E.RestL" }} 
+										{{ localize "DND5E.RestL" }}
```
Comment (szefo09) [inline src/templates/actors/tidy5e-sheet.html:322]: Don't change End of Line Sequence here.

File: src/templates/actors/tidy5e-sheet.html [hunk 10]
```diff
@@ -349,7 +349,7 @@ <h4 class="resource-name">
 
 			{{!-- Inventory Tab --}}
 			<div class="tab inventory {{#if actor.flags.tidy5e-sheet.inventory-grid}}grid{{/if}}" data-group="primary" data-tab="inventory">
-				{{#if actor.flags.tidy5e-sheet.inventory-grid}} 
+				{{#if actor.flags.tidy5e-sheet.inventory-grid}}
```
Comment (szefo09) [inline src/templates/actors/tidy5e-sheet.html:352]: Don't change End of Line Sequence here.

File: src/templates/actors/tidy5e-sheet.html [hunk 11]
```diff
@@ -358,7 +358,7 @@ <h4 class="resource-name">
 
 			{{!-- Spellbook Tab --}}
 			<div class="tab spellbook {{#if actor.flags.tidy5e-sheet.spellbook-grid}}grid{{/if}}" data-group="primary" data-tab="spellbook">
-				{{#if actor.flags.tidy5e-sheet.spellbook-grid}} 
+				{{#if actor.flags.tidy5e-sheet.spellbook-grid}}
```
Comment (szefo09) [inline src/templates/actors/tidy5e-sheet.html:361]: Don't change End of Line Sequence here.

**Review by szefo09 — CHANGES_REQUESTED** [review]
It's good effort to compile the changes, but I don't believe this is a complite update to the DnD5e 2.1.2 system. There are still issues that aren't solved by this PR like #626 and #627 . And #627 is also very destructive, as it can cause players to lose all of their notes.

File: src/scripts/tidy5e-npc.js (hunk 1 above)
Comment (aquavitae) [reply src/scripts/tidy5e-npc.js:33]: This isn't an EOL change, my IDE strips trailing whitespace (same with tabulation below). I thought I'd skipped them when committing but it seems some slipped through.

**Comment by pccool007** [comment]
When will it be merge ?

**Comment by aquavitae** [comment]
I think #622 is more complete than this. I've left this one open for now but I'll probably close it soon.

**Pull request closed without merging**